crates/dojo-lang/src/commands/delete.rs

```diff
@@ -0,0 +1,90 @@
+use std::collections::HashMap;
+
+use cairo_lang_defs::plugin::PluginDiagnostic;
+use cairo_lang_semantic::patcher::RewriteNode;
+use cairo_lang_syntax::node::db::SyntaxGroup;
+use cairo_lang_syntax::node::{ast, Terminal, TypedSyntaxNode};
+
+use super::{CommandData, CommandTrait};
+
+#[derive(Clone)]
+pub struct DeleteCommand {
+    data: CommandData,
+    pub components: Vec<smol_str::SmolStr>,
+}
+
+impl DeleteCommand {
+    fn handle_struct(&mut self, db: &dyn SyntaxGroup, query: ast::Arg, expr: ast::Expr) {
+        if let ast::Expr::StructCtorCall(ctor) = expr {
```

Comment:
you'll have to change this logic to look similar to the `entity` commands. it'll need to use https://github.com/dojoengine/dojo/blob/9eaa1a76f74f7268bccee811b31497f55395e1f8/crates/dojo-lang/src/commands/entities.rs#L159 to get the components from the generic args